PHC-Android/PHC/src/main/java/phc/android/Checkout/CheckoutConfirmationFragment.java

```diff
@@ -147,11 +201,96 @@ public void onSaveInstanceState(Bundle outState) {
      */
     @Override
     protected void confirm() {
-        //recordScan();
-        FragmentTransaction transaction = getActivity().getFragmentManager().beginTransaction();
-        SuccessFragment successFragment = new SuccessFragment();
-        successFragment.setType(SuccessFragment.SuccessType.CHECKOUT_SUCCESS);
-        transaction.replace(R.id.checkout_activity_container, successFragment);
-        transaction.commit();
+        fillFields();
+
+    }
+
+    /**
+     * Creates hashmap that has checkout fields (comments, experience, services not received
+     */
+    private void fillFields(){
+       HashMap<String, Object> fields = getFields();
+        mUserPreferences = getActivity().getSharedPreferences(USER_AUTH_PREFS_NAME,
+                Context.MODE_PRIVATE);
+        String userId = mUserPreferences.getString("user_id", null);
+        String authToken = mUserPreferences.getString("auth_token", null);
+
+        sRequestManager.requestUpdateFeedback(
+                fields,
+                userId,
+                authToken,
+                new UpdateResponseListener(),
+                new UpdateErrorListener());
+    }
+
+    private class UpdateResponseListener implements Response.Listener<JSONObject>{
+
+        @Override
+        public void onResponse(JSONObject jsonObject){
+            //mUserInfo.edit().clear().apply();
+            FragmentTransaction transaction = getActivity().getFragmentManager().beginTransaction();
+            SuccessFragment successFragment = new SuccessFragment();
+            successFragment.setType(SuccessFragment.SuccessType.CHECKOUT_SUCCESS);
+            transaction.replace(R.id.checkout_activity_container, successFragment);
+            transaction.commit();
+
+
+            Log.d(TAG, jsonObject.toString());
+        }
+    }
+    private class UpdateErrorListener implements Response.ErrorListener{
+        @Override
+        public void onErrorResponse(VolleyError volleyError){
+            if (volleyError != null && volleyError.getLocalizedMessage()!=null) {
+                Log.e(TAG, "volleyError.getLocalizedMessage() " + volleyError.getLocalizedMessage());
+                volleyError.printStackTrace();
+            }
+            Toast toast = Toast.makeText(getActivity(), "Error checking out user", Toast.LENGTH_SHORT);
+            toast.show();
+        }
     }
-}
\ No newline at end of file
+
+    /**
+     * A Helper function for getting the relevant information to post
+     * @return a Map containing key value pairs of checkout information. The keys are
+     * field names, and the values are their associated values.
+     */
```

Comment:
yah i see, but I think it's better if we stick to the same convention as the other network requests.
as in, you should do:

sRequestManager.requestUpdateFeedback(
		mComments,
		mExperience,
		mServicesNotReceived,
		mScanResult, 
		userId,
		authToken,
		new UpdateResponseListener(),
		new UpdateErrorListener());